walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -35,20 +35,48 @@
   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     Set<String> desiredGuests = new HashSet<>();
+    Set<String> possibleGuests = new HashSet<>();
     desiredGuests.addAll(request.getAttendees());
+    possibleGuests.addAll(request.getOptionalAttendees());
     int duration = (int) request.getDuration();
 
     if (duration > LATEST_TIME){
       return new ArrayList<>();
     }
     if (desiredGuests.size() == 0){
-      ArrayList<TimeRange> freeDay = new ArrayList<>();
-      freeDay.add(TimeRange.WHOLE_DAY);
-      return freeDay;
+      if (possibleGuests.size() == 0){
+        ArrayList<TimeRange> freeDay = new ArrayList<>();
+        freeDay.add(TimeRange.WHOLE_DAY);
+        return freeDay;
+      } else {
+        return checkOptionalAttendance(possibleGuests, duration, events, null);
+      }
     }
     int start = 0;
 
     // Find all the times when our desired attendees are busy
+    ArrayList<ArrayList<TimeRange>> busyTimes = findBusy(desiredGuests, events);
+    
+    // Use our busy times list to find all times when our guests are NOT busy
+    ArrayList<ArrayList<TimeRange>> allTimes = new ArrayList<>();
+    for (ArrayList<TimeRange> guestSchedule : busyTimes){
+      allTimes.add(findAvailability(guestSchedule, duration));
+    }
+    ArrayList<TimeRange> mandatoryGuests = findIntersectionBetweenLists(allTimes, duration);
```

Comment:
maybe a better name for this variable, because it's a list of available time, not a list of guests.